### PR TITLE
Select a newly created item in the list and clear the filter in the dxSelectBox after item added (T699477)

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

File: js/ui/select_box.js [hunk 1]
```diff
@@ -778,11 +778,11 @@ var SelectBox = DropDownList.inherit({
 
         item = item || null;
         this.option("selectedItem", item);
-        this._setValue(this._valueGetter(item));
-        this._renderDisplayText(this._displayGetter(item));
-        if(item === null && this._wasSearch()) {
+        if(this._wasSearch()) {
```
Comment (lukyanovas) [inline js/ui/select_box.js]: Check TagBox behavior

Comment (lukyanovas) [reply js/ui/select_box.js]: _shouldClearFilter ?

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -2178,6 +2178,31 @@ QUnit.test("search timer should not be cleared when the widget is opening", func
     assert.equal($(selectBox.content()).find(toSelector(LIST_ITEM_CLASS)).length, 3, "filter was cleared");
 });
 
+QUnit.test("Custom value should be selected in the list", function(assert) {
```
Comment (churkin) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Rewrite to ES6 please

**Review by churkin — CHANGES_REQUESTED** [review]

**Review by churkin — DISMISSED** [review]

**Review by lukyanovas — CHANGES_REQUESTED** [review]
If you want to use ES6 in tests, rewrite all test file(or `import` at least)

**Review by churkin — APPROVED** [review]

**Review by lukyanovas — APPROVED** [review]

**Pull request merged**